starfish/pipeline/features/spots/detector/local_max_peak_finder.py

```diff
@@ -112,26 +119,29 @@ def find_attributes(self, image: np.ndarray) -> SpotAttributes:
         attributes['spot_id'] = np.arange(attributes.shape[0])
         return SpotAttributes(attributes)
 
-    def find(self, stack: ImageStack):
+    def find(self, data_stack: ImageStack, blobs_image: Union[np.ndarray, xr.DataArray]=None) \
+            -> IntensityTable:
         """
         Find spots.
 
         Parameters
         ----------
-        stack : ImageStack
+        data_stack : ImageStack
             Stack where we find the spots in.
-        """
-        spot_attributes: List[SpotAttributes] = stack.transform(
-            self.find_attributes, is_volume=self.is_volume, verbose=self.verbose)
-
-        # TODO ambrosejcarr: do we need to find spots in the aux_dict too?
+        blobs_image : Union[np.ndarray, xr.DataArray]
+            If provided, spots will be found in the blobs image, and intensities will be measured
+            across hybs and channels. Otherwise, spots are measured independently for each channel
+            and round.
 
-        # TODO ambrosejcarr: this is where development stopped; spot_attributes is correct, but translating
-        # spot_attributes into an encoder_table is tricky without first implementing the new codebook. Do that first.
-        # create an encoded table
-        # encoded_spots = self.encode(spot_attributes.data)
+        """
+        intensity_table = detect_spots(
```

Comment:
i think this should also allow the user the ability to compute the blobs image as the max projection, thus fully leveraging detect_spots flexibility.

Reply:
still missing the pass thru of ```reference_image_from_max_projection``` here from constructor to detect_spots